src/main/java/com/weddineo/firebase/service/FirebaseAppInitializer.java

```diff
@@ -0,0 +1,49 @@
+package com.weddineo.firebase.service;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.weddineo.firebase.exception.FirebaseRuntimeException;
+import lombok.extern.log4j.Log4j2;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+@Component
+@Slf4j
+public class FirebaseAppInitializer {
+
+    @Value("${firebase.private.key.path}")
+    private String firebasePrivateKeyPath;
+
+    @Value("${firebase.datebase.url}")
+    private String firebaseDatabaseURL;
+
+    @PostConstruct
+    public void init(){
+        log.info("Initializing firebase service...");
+        try {
+            FileInputStream serviceAccount =
+                    new FileInputStream(firebasePrivateKeyPath);
+
+            FirebaseOptions options = FirebaseOptions.builder()
+                    .setCredentials(GoogleCredentials.fromStream(serviceAccount))
+                    .setDatabaseUrl(firebaseDatabaseURL)
+                    .build();
+
+            if (FirebaseApp.getApps().isEmpty()) {
```

Comment:
I think creating _serviceAccount_ and _options_ could be done in this if block -> optimalization 🤣. And also these 3 operations (creating variables and initializing App) could be extracted to the private method.

Reply:
Done.